### ISSUE TITLE
LV error on Ubuntu 16.04

### ISSUE THREAD

**Issue by deHoeninger** [duplicati/duplicati#2862]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.2.12
- **Operating system**: Ubunutu 16.04
- **Backend**: ACD

## Description

Since a few days, one of my six backup is throwing the following Logical Volume error.
## Steps to reproduce

Happens during normal scheduled backups.

- **Actual result**:
Error Message
- **Expected result**:
No Error Message

## Debug log

```
Failed: Skript wurde erfolgreich abgeschlossen, aber der temporäre Ordner /tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/ existiert weiterhin: Exit code: 0
Error: umount -f "/tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/" failed!
Waiting 5 seconds before retry
  Logical volume "duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106" successfully removed
Error: lvremove failed: lvremove --force "vg0/root"

umount: /tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/: not mounted
umount: /tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/: not mounted
  Logical volume vg0/root contains a filesystem in use.

Details: System.Exception: Skript wurde erfolgreich abgeschlossen, aber der temporäre Ordner /tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/ existiert weiterhin: Exit code: 0
Error: umount -f "/tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/" failed!
Waiting 5 seconds before retry
  Logical volume "duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106" successfully removed
Error: lvremove failed: lvremove --force "vg0/root"

umount: /tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/: not mounted
umount: /tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/: not mounted
  Logical volume vg0/root contains a filesystem in use.

  at Duplicati.Library.Snapshots.LinuxSnapshot.Dispose () <0x4130c000 + 0x0012f> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.BackupHandler.Run (System.String[] sources, IFilter filter) <0x412a65d0 + 0x0187f> in <filename unknown>:0
```

**Comment by davegold24** [comment]
@deHoeninger You don't note, but I assume, that you have snapshot=Auto or On set? Just to verify, if you have snapshots disabled, can you verify that this doesn't occur?

**Comment by deHoeninger** [comment]
Indeed, I'm running all my backups with snapshot-policy = on, but only one of them is currently failing with the above mentioned error message.

**Comment by dgold9670** [comment]
This looks to me like the failure is before this log snippet. Do you see errors in your OS messages log regarding snapshot creation?

I know that most vendors use of Linux snapshots are new, and figuring out what failed would make it easier to add in better error detection.

In this case, "umount -f "/tmp/duplicati-b7369c79-9d9b-4cd0-8c9b-4aea87e4a106/" failed!" seems to indicate either that it was never mounted (most likely) or that some other process is using it (less likely, given other snapshots are working)

Send a df -h output and indicate which one is failing. I wonder if the failure is due to a size issue.

I have to research how Linux LVM snapshots work a bit more in depth, but that'll at least get us moving along on troubleshooting this.

**Comment by dgold9670** [comment]
@deHoeninger If you can run this with logging enabled and debug logging enabled, and send me the output, I'd be glad to review it. I'm thinking more and more than the lvm snapshot failed, and the code didn't catch or clean it up properly.
But without logs, thats just supposition.

**Comment by deHoeninger** [comment]
Hi,
I dig a little bit deeper in that topic and could fix it now.

Simple reason for my issue was, that the lvm snapshot for that backup-job was hitting its size-limit and the lvm auto-growth feature was not enabled.

In the end this was caused by a cron job, which was dumping my databases at the some time my backup job was running. So the snapshot filled up with massive data. Never happend before, cause before the DST change a few weeks ago, the database backup cronjob was running an hour before the duplicati backup job normally starts. Now both have been started at the some time, cause the Duplicati Scheduler is not honoring the DST like the cron system does.

Best

**Comment by davegold24** [comment]
@deHoeninger Very interesting, and good troubleshooting. Would you mind posting a synopsis in the forums, and closing this issue?
